### PR TITLE
Feature improvement/using backendUrl through env for best practice

### LINKED ISSUE #88: Cors Error (Important & Urgent to fix)
We are again getting a Cors error on the deployed URL as well as the deploy preview.
Please fix this issue as urgent as possible.

### PR DESCRIPTION
**Pull request by yours7himanshu**
this fixes #88 
I had include .env file for accessing the backendURL when hitting the api and i am uploading that env file in github just once so that u got to know whats going on and how it worked...even though we won't upload env file because it consist some important info that we want to hide...So make super sure that on next commit u will hide this .env file using the .gitignore

i had imported that backEnd url in code using  one line of code -- const backendURL="import.meta.env.VITE_BACkEND_URL"
You will came to know when u will see it and by using this practice its almost 100%chances that cors problem will not occur
Screenshot from 2024-10-05 18-45-41

### CODE REVIEW

File: frontend/src/Pages/Landing.jsx [hunk 1]
```diff
@@ -7,12 +7,14 @@ import Cards from "../Components/Cards";
 import massa from "../assets/Images/Massaman.png";
 import { useNavigate } from "react-router-dom";
 import axios from "axios";
-import backendURL from "../../common/backendUrl";
+//import backendURL from "../../common/backendUrl";
```
Comment (AlfiyaSiddique) [inline frontend/src/Pages/Landing.jsx:10]: remove this line do not comment out

File: frontend/src/Pages/Landing.jsx [hunk 2]
```diff
@@ -7,12 +7,14 @@ import Cards from "../Components/Cards";
...
 
 const Landing = () => {
   const navigator = useNavigate();
   const [best, setBest] = useState([])
   const [loading, setLoading] = useState(true);
+   const backendURL = import.meta.env.VITE_BACKEND_URL;
+   //console.log(backendURL);
```
Comment (AlfiyaSiddique) [inline frontend/src/Pages/Landing.jsx:17]: remove this line

File: frontend/src/Pages/Landing.jsx [hunk 3]
```diff
@@ -36,16 +38,22 @@ const Landing = () => {
...
     axios.get(`${backendURL}/api/recipes`)
-    .then((res)=>{
-      setBest(res.data.recipes.slice(0,3))
-    }).catch((err)=>{
-      console.log(err)
-    }).finally(()=> setLoading(false))
-  }, [best, setBest])
-  
+        .then((res) => {
+            console.log(res.data); // Log the response data
```
Comment (AlfiyaSiddique) [inline frontend/src/Pages/Landing.jsx:46]: remove this console

File: frontend/src/Pages/Recipes.jsx [hunk 4]
```diff
@@ -1,6 +1,6 @@
 // eslint-disable-next-line no-unused-vars
 import React, { useState, useEffect } from "react";
-import backendURL from "../../common/backendUrl";
+//import backendURL from "../../common/backendUrl";
 import axios from "axios";
```
Comment (AlfiyaSiddique) [inline frontend/src/Pages/Recipes.jsx:4]: remove comment

**Comment by AlfiyaSiddique** [comment]
@yours7himanshu some comments please fix them

**Comment by AlfiyaSiddique** [comment]
Also Are you sure that the landing and recipe are the only pages which need backendURL?
I searched it and found this
image
Can you also search and confirm

**Comment by yours7himanshu** [comment]
> Also Are you sure that the landing and recipe are the only pages which need backendURL? I searched it and found this image Can you also search and confirm

No we need backendUrl in every file in which making request to the backend server... Recently I saw landing.jsx and recipe.jsx file using backend Url to hit api endpoint we had created so I had changed on that file currently

**Comment by AlfiyaSiddique** [comment]
Okay then please change in all files

**Comment by yours7himanshu** [comment]
Okay mam sure

**Review by yours7himanshu — COMMENTED** [review]
Removed the comments as u had told

**Comment by yours7himanshu** [comment]
> Okay then please change in all files

So should I change then code in file and then raise a PR again?

**Comment by AlfiyaSiddique** [comment]
No chain in this branch only. The change will come under this PR only

**Comment by yours7himanshu** [comment]
> No chain in this branch only. The change will come under this PR only

Okay thnks for the clarification

**Comment by yours7himanshu** [comment]
> No change in this branch only. The change will come under this PR only

Completed changes in all files and removed the desired comments that u had told pls review it again...I had recently committed

**Comment by AlfiyaSiddique** [comment]
Will review in sometime

**Comment by yours7himanshu** [comment]
> Will review in sometime

Ya sure mam and let me know what next I have to do😊

**Comment by AlfiyaSiddique** [comment]
@yours7himanshu navbar.jsx also has backendURl use did you change that? Please check once.

**Comment by yours7himanshu** [comment]
> @yours7himanshu navbar.jsx also has backendURl use did you change that? Please check once.

Ohh damm I forgot to open component folder 😂. that's the reason didi I didn't change Nav.jsx backend Url😅

**Comment by yours7himanshu** [comment]
> @yours7himanshu navbar.jsx also has backendURl use did you change that? Please check once.

Now didi all complete u can review now

**Review by AlfiyaSiddique — APPROVED** [review]

**Comment by AlfiyaSiddique** [comment]
Merging, Looks good
Thanks @yours7himanshu for the contribution
Also please call me by my name.

**Comment by yours7himanshu** [comment]
> Merging, Looks good Thanks @yours7himanshu for the contribution Also please call me by my name.

Your welcome Alfiya....I had DM you in discord please accept my DM and pls send a friend request...I wanted to collaborate with u more....Pls remember my name I will raise new issue according to the need of the project😊

**Comment by yours7himanshu** [comment]
> Merging, Looks good Thanks @yours7himanshu for the contribution Also please call me by my name.

And one thing more please make sure on the next commit using .gitignore don't let that env file pushed into GitHub...env files consist sensitive data it is not meant for sharing.

**Pull request merged**